encode_utils/MetaDataRegistration/eu_register.py

```diff
@@ -150,15 +159,33 @@ def main():
     conn.set_submission(True)
     infile = args.infile
     patch = args.patch
+    rmpatch = args.rm_patch
+    props_to_remove = args.remove_property
+
     gen = create_payloads(profile_id=profile_id, infile=infile)
     for payload in gen:
-        if not patch:
+        if not patch and not rmpatch:
             conn.post(payload, require_aliases=not no_aliases)
-        else:
+        elif rmpatch:
+            if not props_to_remove:
+                raise Exception("No properties to remove were specified. Use --patch if only patching is needed.")
+            else:
+                props_to_remove = props_to_remove.split(',')
+            record_id = payload.get(RECORD_ID_FIELD, False)
+            if not record_id:
+                raise Exception(
+                    "Can't patch payload {} since there isn't a '{}' field indicating an identifier for the record to be PATCHED.".format(
+                        euu.print_format_dict(payload), RECORD_ID_FIELD))
+            payload.pop(RECORD_ID_FIELD)
+            payload.update({conn.ENCID_KEY: record_id})
+            conn.remove_and_patch(props=props_to_remove, patch=payload, extend_array_values=not overwrite_array_values)
+        elif props_to_remove and not rmpatch:
+            raise Exception("Properties to remove were specified, but --rm-patch flag was not set.")
```

Comment:
Similar to an above comment. This code should be moved up to right after the `parser.parse_args()`, and use parser.error:
```suggestion
        if args. props_to_remove and not args.rmpatch:
            parser.error("Properties to remove were specified, but --rm-patch flag was not set.")
```